graylog2-web-interface/src/components/search/MessageFieldExtractorActions.jsx

```diff
@@ -25,17 +25,22 @@ const MessageFieldExtractorActions = React.createClass({
     );
   },
   render() {
-    return (
-      <div className="message-field-actions pull-right">
-        <DropdownButton pullRight
-                     bsSize="xsmall"
-                     title="Select extractor type"
-                     key={1}
-                     id={`select-extractor-type-dropdown-field-${this.props.fieldName}`}>
-          {ExtractorUtils.EXTRACTOR_TYPES.map(extractorType => this._formatExtractorMenuItem(extractorType))}
-        </DropdownButton>
-      </div>
-    );
+    let messageField = this.props.message.fields[this.props.fieldName];
+    if(typeof messageField === 'string') {
+      return (
+          <div className="message-field-actions pull-right">
+            <DropdownButton pullRight
+                            bsSize="xsmall"
+                            title="Select extractor type"
+                            key={1}
+                            id={`select-extractor-type-dropdown-field-${this.props.fieldName}`}>
+                {ExtractorUtils.EXTRACTOR_TYPES.map(extractorType => this._formatExtractorMenuItem(extractorType))}
+            </DropdownButton>
+          </div>
+        );
+    } else {
+      return (<div className="message-field-actions pull-right"/>);
```

Comment:
✅ Personally, I find it more obvious if the dropdown list was simply missing, but for the sake of consistency I've added back the dropdown and disabled it for non-string fields.